cfn-resources/federated-database-instance/cmd/resource/resource.go

```diff
@@ -69,16 +68,19 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 	if peErr != nil {
 		return *peErr, nil
 	}
-	createFederatedDatabaseAPIRequest := client.Atlas20231115002.DataFederationApi.CreateFederatedDatabase(context.Background(), *currentModel.ProjectId, &dataLakeTenantInput)
-	dataLakeTenant, response, err := createFederatedDatabaseAPIRequest.Execute()
+	dataLakeTenant, response, err := client.AtlasSDK.DataFederationApi.CreateFederatedDatabaseWithParams(
+		context.Background(),
+		&atlasSDK.CreateFederatedDatabaseApiParams{
+			GroupId:        *currentModel.ProjectId,
+			DataLakeTenant: &dataLakeTenantInput,
+		}).Execute()
 
 	defer closeResponse(response)
```

Comment:
uhm, this defer seems weird, do we need it?

Reply:
I tried to change as little as possible the existing code. Looking at the http.Response entity, you can see:
_"The http Client and Transport guarantee that Body is always non-nil, even on responses without a body or responses with a zero-length body. It is the caller's responsibility to close Body. The default HTTP client's Transport may not reuse HTTP/1.x "keep-alive" TCP connections if the Body is not read to completion and closed."_

Reply:
i understand but also we have to be good boyscouts :-)

we never close http responses for SDK API calls, I'd like to understand why it's different here

Reply:
Went deeper into it. Basically all clients have to close the body if you want to [avoid resource leaks](https://stackoverflow.com/questions/33238518/what-could-happen-if-i-dont-close-response-body). The good thing is that the autogenerated SDK closes the body after request ha been done. I removed it from this resource.